### PR TITLE
chore: webhook functions now return AdmissionResponse

### LINKED ISSUE #1267: pkg/controller/webhook.Validate et al should return an `AdmissionResponse`
**What would you like to be added**:
The functions in [pkg/controller/webhook/util.go](https://github.com/kubernetes-sigs/kubefed/blob/bb3cd73f60ec8053c7fb991c94efe6d4a99aaced/pkg/controller/webhook/util.go) should not mutate the `admission.Response` parameter but rather return a new one.

**Why is this needed**:
This came out of [this PR review](https://github.com/kubernetes-sigs/kubefed/pull/1263#discussion_r465677103).

/kind feature

### PR DESCRIPTION
**Pull request by makkes**
**What this PR does / why we need it**:

Having these functions modify the AdmissionResponse passed in as
parameter makes it hard to reason about the code.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1267 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/controller/webhook/util.go [hunk 1]
```diff
@@ -56,43 +56,52 @@ func NewValidatingResource(resourcePluralName string) schema.GroupVersionResourc
...
-		status.Allowed = true
-		return true
+		return &admission.Response{
+			AdmissionResponse: admissionv1beta1.AdmissionResponse{
+				Allowed: true,
+			},
+		}
 	}
-	return false
+	return nil
```
Comment (RainbowMango) [inline pkg/controller/webhook/util.go:72]: I guess return `nil` equals to `not allowed` would bring ambiguous, and return a `admission.Response` will break the function as well.

The function name `Allowed` designed to check whether a specific request is allowed. So the return type should be `bool`.

A simple way maybe just remove `status` from the original function signature.

Comment (makkes) [reply pkg/controller/webhook/util.go:72]: Thanks for the feedback, @RainbowMango, I'll look to improve the signature of this func.

**Comment by makkes** [comment]
I'll have to have a look at the failing tests as well.

**Comment by makkes** [comment]
Ok, failing test was just caused by the linter.

File: pkg/controller/webhook/util.go (hunk 1 above)
Comment (makkes) [reply pkg/controller/webhook/util.go]: @RainbowMango I addressed your concern and improved the utility functions a bit. PTAL.

**Comment by RainbowMango** [comment]
I guess commit should be squashed, otherwise, LGTM.

**Comment by makkes** [comment]
commits are squashed now.

**Comment by RainbowMango** [comment]
/lgtm
Thanks.

**Comment by jimmidyson** [comment]
/approved

Thanks @makkes!

**Comment by jimmidyson** [comment]
/approve

Thanks @makkes!

**Pull request merged**